flutter_mobx_forms/lib/src/form_field.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:mobx/mobx.dart';
+
+part 'form_field.g.dart';
+
+typedef SyncFieldValidator<T> = Iterable<String> Function(T value);
+typedef AsyncFieldValidator<T> = Stream<String> Function(T value);
+
+enum ValidationPolicy { onChange, always, manual }
+
+class FormField<T> = _FormField<T> with _$FormField;
+
+abstract class _FormField<T> with Store {
+  _FormField(
+      {this.name,
+      this.label,
+      this.value,
+      this.validator,
+      this.asyncValidator,
+      this.validationPolicy = ValidationPolicy.manual}) {
+    // ignore: prefer_asserts_in_initializer_lists
+    assert(validator != null && asyncValidator != null,
+        'Only one of validator or asyncValidator can be specified');
+  }
+
+  ValidationPolicy validationPolicy;
+
+  final String name;
+
+  final String label;
+
+  @observable
+  List<String> errors;
+
+  @observable
+  T value;
+
+  @observable
+  bool _isValidating;
+
+  @computed
+  bool get isValidating => _isValidating;
+
+  @computed
+  String get error =>
+      errors == null ? null : (errors.isEmpty ? null : errors[0]);
+
+  @computed
+  bool get isValid => errors == null;
+
+  final SyncFieldValidator<T> validator;
+  final AsyncFieldValidator<T> asyncValidator;
+
+  void validate() {
+    if (validator != null) {
```

Comment:
Should we prevent validation inside validation?

```suggestion
    assert(!_isValidating);
    if (validator != null) {
```